### PR TITLE
remove scheduler component config v1beta1

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
As component config will be evolved to `v1beta3` in release1.23, we will remove `v1beta1` support as maintaining 3+  apis seems unnecessary.

#### Which issue(s) this PR fixes:

Fixes #https://github.com/kubernetes/enhancements/issues/2901

#### Does this PR introduce a user-facing change?

```release-note
kube-scheduler: support for configuration file version `v1beta1` is removed. Update configuration files to v1beta2(xref: https://github.com/kubernetes/enhancements/issues/2901) or v1beta3 before upgrading to 1.23.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/pull/2921
```

### CODE REVIEW

File: test/integration/scheduler/queue_test.go [hunk 1]
```diff
@@ -49,127 +47,6 @@ import (
 	"k8s.io/utils/pointer"
 )
 
-func TestServiceAffinityEnqueue(t *testing.T) {
```
Comment (ahg-g) [inline test/integration/scheduler/queue_test.go:52]: This test was added to test event registration.

@Huang-Wei is this covered somewhere else? ServiceAffinity is not supported in v1beta2 and above.

**Review by ahg-g — COMMENTED** [review]
you also need to remove the definitions in the staging repo: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/kube-scheduler/config/v1beta1

File: pkg/scheduler/apis/config/validation/validation.go [hunk 2]
```diff
@@ -147,12 +142,6 @@ var removedPluginsByVersion = []removedPlugins{
 // Remember to add an entry to that list when creating a new component config
 // version (even if the list of conflict plugins is empty).
 var conflictScorePluginsByVersion = map[string]map[string]sets.String{
-	v1beta1.SchemeGroupVersion.String(): {
```
[the remark this answers is not in the document]
Comment (kerthcet) [reply pkg/scheduler/apis/config/validation/validation.go:150]: > you also need to remove the definitions in the staging repo: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/kube-scheduler/config/v1beta1

gotcha, solved my confusion.

File: test/integration/scheduler/queue_test.go (hunk 1 above)
Comment (kerthcet) [reply test/integration/scheduler/queue_test.go:52]: yeah, `ServiceAffinity` was related to several packages, I was plan to remove it in another pr, this pr is also good. Listening to your guys for some advices.

**Comment by ahg-g** [comment]
@kerthcet is this PR still a draft? if not, please remove the label so we can tag it for api-review.

**Comment by ahg-g** [comment]
looks good to me apart from the comment I left for @Huang-Wei.

File: test/integration/scheduler/queue_test.go (hunk 1 above)
Comment (Huang-Wei) [reply test/integration/scheduler/queue_test.go:52]: > @Huang-Wei is this covered somewhere else?

Nope, this is the only integration test to cover event registration (for core API resource).

We should replace it with another test case, like a volume event to trigger moving pods that failed the VolumeBinding plugin. Given it's irrelevant with this PR, just leave a TODO to me. SG?

```
// TODO(Huang-Wei): Add a test case to cover event registration for core API resources.
```

Comment (ahg-g) [reply test/integration/scheduler/queue_test.go:52]: Sounds good to me, but lets create an issue to track it and assign the todo to the issue.

Comment (kerthcet) [reply test/integration/scheduler/queue_test.go:52]: what should we do:
1. add a comment above
2. open a new pr to track the removal of ServiceAffinity
3. Add another test case as #105303 describes.

or we should combine `2` and `3` together in one pr?

**Comment by kerthcet** [comment]
> @kerthcet is this PR still a draft? if not, please remove the label so we can tag it for api-review.

opened already.

File: test/integration/scheduler/queue_test.go [hunk 3]
```diff
@@ -49,126 +47,7 @@ import (
...
-		t.Fatalf("Failed to delete service 'svc': %v", err)
-	}
-
-	// Now we should be able to pop the Pod from activeQ again.
-	podInfo = nextPodOrDie(t, testCtx)
-	if podInfo.Attempts != 2 {
-		t.Errorf("Expected the Pod to be attempted 2 times, but got %v", podInfo.Attempts)
-	}
-}
+// TODO(Huang-Wei): Add a test case to cover event registration for core API resources, refer to issue:https://github.com/kubernetes/kubernetes/issues/105303
```
Comment (Huang-Wei) [inline test/integration/scheduler/queue_test.go:50]:
```suggestion
// TODO(#105303): Add a test case to cover event registration for core API resources
```

File: test/integration/scheduler/queue_test.go (hunk 1 above)
Comment (Huang-Wei) [reply test/integration/scheduler/queue_test.go:52]: > or we should combine `2` and `3` together in one pr?

In one pr please.

File: test/integration/scheduler/queue_test.go (hunk 3 above)
Comment (kerthcet) [reply test/integration/scheduler/queue_test.go]: done :)

**Comment by kerthcet** [comment]
kindly ping @Huang-Wei @ahg-g @damemi

**Comment by Huang-Wei** [comment]
/approve
on scheduling side.

**Comment by kerthcet** [comment]
I was wondering whether someone I can assign to help reviewing this PR to prevent blocking another related one.

**Comment by liggitt** [comment]
/approve

please update the release note with a link to the docs for the v1beta1 → v1beta2 (or v1beta3) migration instructions

**Comment by pacoxu** [comment]
/release-note-edit
```release-note
kube-scheduler: support for configuration file version `v1beta1` is removed. Update configuration files to v1beta2(xref: https://github.com/kubernetes/enhancements/issues/2901) or v1beta3 before upgrading to 1.23.
```

**Pull request merged**